lib/rubygems/commands/update_command.rb

```diff
@@ -209,6 +211,16 @@ def preparing_gem_layout_for(version)
     end
   end
 
+  def preparing_default_gemspecs_for(version)
+    if Gem::Version.new(version) >= Gem::Version.new("3.3.a")
```

Comment:
Actually I can see now why the gemspecs were not being regenerated back when downgrading in my testing: master is also `>= 3.3.a`. But I think this might not be actually needed because the extra comment doesn't seem to cause any issues when running old versions.